tfe/credentials.go

```diff
@@ -0,0 +1,101 @@
+package tfe
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	svchost "github.com/hashicorp/terraform-svchost"
+	"github.com/hashicorp/terraform-svchost/disco"
+)
+
+func collectCredentialsFromEnv() map[svchost.Hostname]string {
+	const prefix = "TF_TOKEN_"
+
+	ret := make(map[svchost.Hostname]string)
+	for _, ev := range os.Environ() {
+		eqIdx := strings.Index(ev, "=")
```

Comment:
Just curious here on readability: Would it be better to `strings.Split()` on `=` and compare each index rather than using ranges ?

Reply:
Split might be slightly more readable, but this version doesn't do any array allocation and is shared with terraform already. I think I'd like to keep it.